ctk/common/src/test/kotlin/org/codice/compilance/verification/core/SamlDefinedIdentifiersVerifierSpec.kt

```diff
@@ -0,0 +1,281 @@
+                    |<samlp:Response
+                    |  xmlns:saml="$ASSERTION_NAMESPACE"
+                    |  xmlns:samlp="$PROTOCOL_NAMESPACE"
+                    |  ID="id"
+                    |  Version="$VERSION"
+                    |  IssueInstant="$now">
+                    |  <samlp:Status>
+                    |    <samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success"/>
+                    |  </samlp:Status>
+                    |  <saml:Assertion ID="id" IssueInstant="$now" Version="$VERSION">
+                    |    <saml:Issuer
+                    |    $extraIdentifierAttribute
+                    |    Format="$identifierFormat">$identifierValue</saml:Issuer>
+                    |    <saml:AttributeStatement>
+                    |      <saml:Attribute Name="$attributeName" NameFormat="$attributeFormat" />
+                    |    </saml:AttributeStatement>
+                    |  </saml:Assertion>
+                    |</samlp:Response>
+                    """.trimMargin())
+        }
+
+        /* 8.2 URI/Basic name attribute formats */
+        "valid 'unspecified' attribute name" {
+            createResponse(
+                    attributeName = "Unspecified Name",
+                    attributeFormat = ATTRIBUTE_NAME_FORMAT_UNSPECIFIED).let {
+                SamlDefinedIdentifiersVerifier(it).verify()
+            }
+        }
+
+        // No invalid test for 'unspecified' format; other than what is allowed in xml
+
+        "valid URI attribute name" {
+            createResponse(
+                    attributeName = "validURI.com",
+                    attributeFormat = ATTRIBUTE_NAME_FORMAT_URI).let {
+                SamlDefinedIdentifiersVerifier(it).verify()
+            }
+        }
+
+        "invalid URI attribute name" {
+            createResponse(
+                    attributeName = "Whitespace Not Allowed In URI",
+                    attributeFormat = ATTRIBUTE_NAME_FORMAT_URI).let {
+                shouldThrow<SAMLComplianceException> {
+                    SamlDefinedIdentifiersVerifier(it).verify()
+                }.message?.shouldContain(SAMLCore_8_2_2_a.message)
+            }
+        }
+
+        "valid Basic attribute name" {
+            createResponse(
+                    attributeName = "BasicName",
+                    attributeFormat = ATTRIBUTE_NAME_FORMAT_BASIC).let {
+                SamlDefinedIdentifiersVerifier(it).verify()
+            }
+        }
+
+        "invalid Basic attribute name" {
+            createResponse(
+                    attributeName = "Whitespace Not Allowed In Attribute Name",
+                    attributeFormat = ATTRIBUTE_NAME_FORMAT_BASIC).let {
+                shouldThrow<SAMLComplianceException> {
+                    SamlDefinedIdentifiersVerifier(it).verify()
+                }.message?.shouldContain(SAMLCore_8_2_3_a.message)
+            }
+        }
+
+        /* 8.3 Name Identifier Format Identifiers */
```

Comment:
✏️ Should read like `/** 8.3.2 Email Address */` to be consistent with the actual class.